packages/common/src/gckms/KeyInjectorPlugin.js

```diff
@@ -0,0 +1,41 @@
+const kms = require("@google-cloud/kms");
+const { Storage } = require("@google-cloud/storage");
+const { extendConfig } = require("hardhat/config");
+const { HardhatPluginError } = require("hardhat/plugins");
+const { getGckmsConfig } = require("./GckmsConfig");
+
+const { GCKMS_KEYS, MNEMONIC } = process.env;
+
+async function retrieveGckmsKeys(gckmsConfigs) {
```

Comment:
So this is getting access to the key in `key management system` and using that to decrypt the file stored in `data storage` that can be found @ `keyMaterialBucket: cipherTextFile`?

A comment above this function would be useful for future maintainers

Reply:
Yeah, it's basically a more streamlined version of what's in `ManagedSecretProvider.js`. I may just combine the implementations so it's not repeated. I originally avoided that to minimize changes in production code, but 🤷 .

Reply:
Done!